include/npcomp/Dialect/ATen/ATenOpInterface.td

```diff
@@ -0,0 +1,57 @@
+//===- ATenOpInterface.td ----------------------------------*- tablegen -*-===//
+//
+// This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+include "mlir/IR/OpBase.td"
+
+#ifndef ATEN_OP_INTERFACES
+#define ATEN_OP_INTERFACES
+
+def StatisticsOpInterface : OpInterface<"StatisticsOpInterface"> {
+  let description = [{
+     This interface allows ops to expose a static operation profile,
+	  describing the computational behavior of their function.
+  }];
+
+  let methods = [
+    InterfaceMethod<
+      "Return statistics about the compute requirements of an op",
+      "std::map<std::string, uint64_t>", "getStatistics"
```

Comment:
This looks like a quite heavy type to return here. 
It isn't clear to me what this is doing right now, but taking a StringMap (or better if possible a `DenseMap<StringRef. uint64_t>`) as output operand would allow to not realloc constantly by reusing the same map over and over from the call site.

Reply:
I think this is a placeholder for more interesting things.  I've added an issue related to this.